### PR TITLE
Redo ServiceAssuranceExtType as annotations

### PR DESCRIPTION
**Pull request by jvasile**
* Remove ServiceAssuranceExtType class
* Remove LU prefix from service assurance ext throughout tree
* Add service assurance ext to seed.sql
* Add service assurance ext to persistence.xml
* Use annotations in service assurance ext java file

This is my first annotation migration PR.  Please speak slowly and use small words.

### CODE REVIEW

File: psm-app/cms-business-process/src/main/resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -58,6 +58,7 @@
     <class>gov.medicaid.entities.RiskLevel</class>
     <class>gov.medicaid.entities.Role</class>
     <class>gov.medicaid.entities.ServiceAssuranceType</class>
+    <class>gov.medicaid.entities.ServiceAssuranceExtType</class>
```
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/resources/META-INF/persistence.xml]: Would you mind sorting these in alphabetical order? It takes away a decision point ("where does this best fit...?"). :smiley:

File: psm-app/services/src/main/java/gov/medicaid/entities/ServiceAssuranceExtType.java [hunk 2]
```diff
@@ -3,40 +3,30 @@
...
+import javax.persistence.Id;
+import javax.persistence.Table;
+
 /**
  * Represents the Assurance Statements lookup for Chemical Dependency Program.
- * 
- * @author cyberjag
- * @version 1.0
+ *
+ * Note that the Chemical Dependency Program is MN-specific.
```
Comment (jasonaowen) [inline psm-app/services/src/main/java/gov/medicaid/entities/ServiceAssuranceExtType.java]: :+1: Good catch! This is a great place to document that.

File: psm-app/data_load/mmis_load_utility/src/META-INF/Medicaid.hbm.xml [hunk 3]
```diff
@@ -673,7 +673,7 @@
 			<column name="PATIENT_IND" />
 		</property>
 	</class>
-	<class name="gov.medicaid.entities.ServiceAssuranceExtType" table="LU_SVC_ASSURANCE_EXT_TYP">
+	<class name="gov.medicaid.entities.ServiceAssuranceExtType" table="SVC_ASSURANCE_EXT_TYP">
```
Comment (jvasile) [inline psm-app/data_load/mmis_load_utility/src/META-INF/Medicaid.hbm.xml]: I just grepped for LU_ prefix and changed everything I saw.  I don't know that everybody has to do it. I'll make the change you suggest.

File: psm-app/cms-business-process/src/main/resources/META-INF/persistence.xml (hunk 1 above)
Comment (jvasile) [inline psm-app/cms-business-process/src/main/resources/META-INF/persistence.xml]: Done.

File: psm-app/services/src/main/java/gov/medicaid/entities/ServiceAssuranceExtType.java [hunk 4]
```diff
@@ -3,40 +3,30 @@
...
-     */
+    @Id
+    @Column(name = "code")
     private String serviceAssuranceCode;
 
-    /**
-     * Empty constructor.
-     */
     public ServiceAssuranceExtType() {
     }
```
Comment (jvasile) [inline psm-app/services/src/main/java/gov/medicaid/entities/ServiceAssuranceExtType.java]: Done.

File: psm-app/data_load/mmis_load_utility/src/META-INF/Medicaid.hbm.xml (hunk 3 above)
Comment (jvasile) [inline psm-app/data_load/mmis_load_utility/src/META-INF/Medicaid.hbm.xml]: Changed to service_assurance_ext_types

File: psm-app/db/pesp-schema-17-jun.sql [hunk 5]
```diff
@@ -90,1595 +90,1595 @@
...
 
-  CREATE TABLE "CMS"."LU_STATE" 
-   (	"CODE" VARCHAR2(2 CHAR), 
+  CREATE TABLE "CMS"."LU_STATE"
+   (	"CODE" VARCHAR2(2 CHAR),
 	"DESCRIPTION" VARCHAR2(255 CHAR)
    ) ;
 --------------------------------------------------------
---  DDL for Table LU_SVC_ASSURANCE_EXT_TYP
+--  DDL for Table SVC_ASSURANCE_EXT_TYP
```
Comment (jvasile) [inline psm-app/db/pesp-schema-17-jun.sql]: Done and done.

File: psm-app/db/FULL-SCHEMA-MAY31.sql [hunk 6]
```diff
@@ -90,1587 +90,1587 @@
...
 
-  CREATE TABLE "CMS"."LU_STATE" 
-   (	"CODE" VARCHAR2(2 CHAR), 
+  CREATE TABLE "CMS"."LU_STATE"
+   (	"CODE" VARCHAR2(2 CHAR),
 	"DESCRIPTION" VARCHAR2(255 CHAR)
    ) ;
 --------------------------------------------------------
---  DDL for Table LU_SVC_ASSURANCE_EXT_TYP
+--  DDL for Table SVC_ASSURANCE_EXT_TYP
```
Comment (jvasile) [inline psm-app/db/FULL-SCHEMA-MAY31.sql]: Done and done.

File: psm-app/db/mpse-clean-setup.sql [hunk 7]
```diff
@@ -90,1595 +90,1595 @@
...
 
-  CREATE TABLE "CMS"."LU_STATE" 
-   (	"CODE" VARCHAR2(2 CHAR), 
+  CREATE TABLE "CMS"."LU_STATE"
+   (	"CODE" VARCHAR2(2 CHAR),
 	"DESCRIPTION" VARCHAR2(255 CHAR)
    ) ;
 --------------------------------------------------------
---  DDL for Table LU_SVC_ASSURANCE_EXT_TYP
+--  DDL for Table SVC_ASSURANCE_EXT_TYP
```
Comment (jvasile) [inline psm-app/db/mpse-clean-setup.sql]: Done and done.

File: psm-app/services/src/main/java/gov/medicaid/entities/ServiceAssuranceExtType.java [hunk 8]
```diff
@@ -3,40 +3,30 @@
...
  */
+@javax.persistence.Entity
+@Table(name = "service_assurance_ext_types")
 public class ServiceAssuranceExtType extends LookupEntity {
 
-    /**
-     * Represents the service assurance code.
-     */
+    @Id
+    @Column(name = "code")
```
Comment (jvasile) [inline psm-app/services/src/main/java/gov/medicaid/entities/ServiceAssuranceExtType.java]: You're right.  The base class handles code.  I've now adjusted it to handle service_assurance_code.

File: psm-app/db/seed.sql [hunk 9]
```diff
@@ -48,6 +49,12 @@ CREATE TABLE service_assurance_types (
   patient_indicator TEXT
 );
 
+CREATE TABLE service_assurance_ext_types (
+  code CHARACTER VARYING(2) PRIMARY KEY,
+  description TEXT UNIQUE,
+  patient_indicator TEXT
```
Comment (jvasile) [inline psm-app/db/seed.sql]: Agreed. Fixed.

**Comment by jvasile** [comment]
@jasonaowen, at this point, I think it's ready for you to look at again

**Pull request merged**